tests/matrix_free/hanging_node_kernels_01.cc

```diff
@@ -0,0 +1,625 @@
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+
+#include <deal.II/matrix_free/fe_evaluation.h>
+#include <deal.II/matrix_free/matrix_free.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+#include "../tests.h"
+
+
+namespace dealii::internal
+{
+  template <int dim, typename Number, bool is_face>
+  struct FEEvaluationImplHangingNodesReference
```

Comment:
Can you add a short documentation this class and in addition, a few lines above, the intent of the present test and what it aims to compare?